From: Simon McVittie <smcv@collabora.com>
Date: Tue, 8 Jan 2019 12:16:37 +0000
Subject: trash test: Don't assume that ~/.local exists

In a minimal autobuilder environment, this test could conceivably be
the first thing to refer to ~/.local.

Modified by Iain Lane <laney@debian.org>: Don't try to create ~/.local
from tests, but skip if it doesn't exist.

Author: Simon McVittie <smcv@collabora.com>
Bug-Upstream: https://gitlab.gnome.org/GNOME/glib/issues/1643
Forwarded: https://gitlab.gnome.org/GNOME/glib/merge_requests/577
Origin: https://gitlab.gnome.org/iainl/glib/commit/aac7b7bbbdc07d306808412f6595b93ad3c9ec8a
---
 gio/tests/trash.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/gio/tests/trash.c b/gio/tests/trash.c
index 1055585..b7df936 100644
--- a/gio/tests/trash.c
+++ b/gio/tests/trash.c
@@ -107,8 +107,19 @@ test_trash_symlinks (void)
 
   g_test_bug ("1522");
 
-  /* The test assumes that ~/.local always exists. */
   target = g_build_filename (g_get_home_dir (), ".local", NULL);
+
+  if (!g_file_test (target, G_FILE_TEST_IS_DIR))
+    {
+      gchar *message;
+
+      message = g_strdup_printf ("Directory '%s' does not exist", target);
+      g_test_skip (message);
+      g_free (message);
+      g_free (target);
+      return;
+    }
+
   target_mount = g_unix_mount_for (target, NULL);
   g_assert_nonnull (target_mount);
   g_test_message ("Target: %s (mount: %s)", target, g_unix_mount_get_mount_path (target_mount));
